File: device-app/android/Java/helloUms/app/src/main/java/com/rationalowl/bridgeUms/PushAppProto.java
package com.rationalowl.bridgeUms;


import com.fasterxml.jackson.annotation.JsonProperty;


public class PushAppProto {

    public static class PushAppAuthNumberReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;

        @JsonProperty("cc")
        public String mCountryCode;
        @JsonProperty("pn")
        public String mPhoneNumber;


        public PushAppAuthNumberReq() {
            mCid = UmsProtocol.APP_AUTH_NUMBER_CMD_ID;
        }
    }


    public static class PushAppAuthNumberRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;
        @JsonProperty("cmt")
        public String mComment;

        public PushAppAuthNumberRes() {
            mCid = UmsProtocol.APP_AUTH_NUMBER_CMD_ID;
        }
    }


    public static class PushAppVerifyAuthNumberReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("dt")
        public int mDeviceType;

        @JsonProperty("pn")
        public String mPhoneNumber;
        @JsonProperty("an")
        public String mAuthNumber;


        public PushAppVerifyAuthNumberReq() {
            mCid = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID;
        }
    }


    public static class PushAppVerifyAuthNumberRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;

        @JsonProperty("cmt")
        public String mComment;

        public PushAppVerifyAuthNumberRes() {
            mCid = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID;
        }
    }


    public static class PushAppInstallReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;

        @JsonProperty("dt")
        public int mDeviceType;
        @JsonProperty("dRId")
        public String mDeviceRegId;

        /* ums fields */
        @JsonProperty("pn")
        public String mPhoneNumber;

        // ums optional
        @JsonProperty("auId")
        public String mAppUserId;
        @JsonProperty("n")
        public String mUserName;


        public PushAppInstallReq() {
            mCid = UmsProtocol.APP_INSTALL_CMD_ID;
        }
    }


    public static class PushAppInstallRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;
        @JsonProperty("cmt")
        public String mComment;

        @JsonProperty("usRid")
        public String mUmsServerRegId;

        public PushAppInstallRes() {
            mCid = UmsProtocol.APP_INSTALL_CMD_ID;
        }
    }


    public static class PushAppUnregUserReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;

        @JsonProperty("dRId")
        public String mDeviceRegId;

        public PushAppUnregUserReq() {
            mCid = UmsProtocol.APP_UNREG_USER_CMD_ID;
        }
    }


    public static class PushAppUnregUserRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;
        @JsonProperty("cmt")
        public String mComment;

        @JsonProperty("dRId")
        public String mDeviceRegId;

        public PushAppUnregUserRes() {
            mCid = UmsProtocol.APP_UNREG_USER_CMD_ID;
        }
    }


    /////// push app job

    public static class PushAppMsgReadNoti {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;

        @JsonProperty("dRId")
        public String mDeviceRegId;
        @JsonProperty("mId")
        public String mMsgId;


        public PushAppMsgReadNoti() {
            mCid = UmsProtocol.APP_MSG_READ_NOTI_CMD_ID;
        }
    }


    public static class PushAppMsgInfoReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("dRId")
        public String mDeviceRegId;
        @JsonProperty("mId")
        public String mMsgId;


        public PushAppMsgInfoReq() {
            mCid = UmsProtocol.APP_MSG_INFO_CMD_ID;
        }
    }


    public static class PushAppMsgInfoRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;
        @JsonProperty("cmt")
        public String mComment;

        @JsonProperty("mId")
        public String mMsgId;
        @JsonProperty("ast")
        public long mAlimtalkSendTime;
        @JsonProperty("ase")
        public int mAlimtalkState;  // 0: unsend, 1: send request, 2: success, 3: fail
        @JsonProperty("mst")
        public long mMunjaSendTime;
        @JsonProperty("mt")
        public int mMunjaType;      // 12: sms, 13: lms, 14: mms
        @JsonProperty("ms")
        public int mMunjaState;     // 0: unsend, 1: send request, 2: success, 3: fail

        public PushAppMsgInfoRes() {
            mCid = UmsProtocol.APP_MSG_INFO_CMD_ID;
        }
    }


    public static class PushAppImgDataReq {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("iId")
        public String mImgId;
        @JsonProperty("mId")
        public String mMsgId;


        public PushAppImgDataReq() {
            mCid = UmsProtocol.APP_IMG_DATA_CMD_ID;
        }
    }


    public static class PushAppImgDataRes {
        @JsonProperty("cId")
        public int mCid;
        @JsonProperty("aId")
        public String mAccountId;
        @JsonProperty("rc")
        public int mResultCode;
        @JsonProperty("cmt")
        public String mComment;

        @JsonProperty("imgD")
        public String mImgData;

        public PushAppImgDataRes() {
            mCid = UmsProtocol.APP_IMG_DATA_CMD_ID;
        }
    }
}

File: device-app/ios-swift/umsDemo/bridgeUms/PushAppProto.swift
public enum PushAppProto {
    public struct PushAppAuthNumberReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mCountryCode = "cc"
            case mPhoneNumber = "pn"
        }

        public var mCid: Int = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        public var mAccountId: String

        public var mCountryCode: String
        public var mPhoneNumber: String
    }

    public struct PushAppAuthNumberRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
        }

        public var mCid: Int = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int
        public var mComment: String?
    }

    public struct PushAppVerifyAuthNumberReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mDeviceType = "dt"
            case mPhoneNumber = "pn"
            case mAuthNumber = "an"
        }

        public var mCid: Int = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        public var mAccountId: String
        public var mDeviceType: Int

        public var mPhoneNumber: String
        public var mAuthNumber: String
    }

    public struct PushAppVerifyAuthNumberRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
        }

        public var mCid: Int = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int

        public var mComment: String?
    }

    public struct PushAppInstallReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mDeviceType = "dt"
            case mDeviceRegId = "dRId"
            case mPhoneNumber = "pn"
            case mAppUserId = "auId"
            case mUserName = "n"
        }

        public var mCid: Int = UmsProtocol.APP_INSTALL_CMD_ID
        public var mAccountId: String

        public var mDeviceType: Int
        public var mDeviceRegId: String
        public var mPhoneNumber: String
        public var mAppUserId: String
        public var mUserName: String
    }

    public struct PushAppInstallRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
            case mUmsServerRegId = "usRid"
        }

        public var mCid: Int = UmsProtocol.APP_INSTALL_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int
        public var mComment: String?

        public var mUmsServerRegId: String?
    }

    public struct PushAppUnregUserReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mDeviceRegId = "dRId"
        }

        public var mCid: Int = UmsProtocol.APP_UNREG_USER_CMD_ID
        public var mAccountId: String

        public var mDeviceRegId: String
    }

    public struct PushAppUnregUserRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
            case mDeviceRegId = "dRId"
        }

        public var mCid: Int = UmsProtocol.APP_UNREG_USER_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int
        public var mComment: String?

        public var mDeviceRegId: String
    }

    /////// push app job

    public struct PushAppMsgReadNoti: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mDeviceRegId = "dRId"
            case mMsgId = "mId"
        }
        public var mCid: Int = UmsProtocol.APP_MSG_READ_NOTI_CMD_ID
        public var mAccountId: String
        public var mDeviceRegId: String
        public var mMsgId: String
    }

    public struct PushAppMsgInfoReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mDeviceRegId = "dRId"
            case mMsgId = "mId"
        }

        public var mCid: Int = UmsProtocol.APP_MSG_INFO_CMD_ID

        public var mAccountId: String
        public var mDeviceRegId: String

        public var mMsgId: String
    }

    public struct PushAppMsgInfoRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
            case mMsgId = "mId"
            case mAlimtalkSendTime = "ast"
            case mAlimtalkState = "ase"
            case mMunjaSendTime = "mst"
            case mMunjaType = "mt"
            case mMunjaState = "ms"
        }

        public var mCid: Int = UmsProtocol.APP_MSG_INFO_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int
        public var mComment: String?

        public var mMsgId: String
        public var mAlimtalkSendTime: Int64
        public var mAlimtalkState: Int // 0: unsend, 1: send request, 2: success, 3: fail
        public var mMunjaSendTime: Int64
        public var mMunjaType: Int // 12: sms, 13: lms, 14: mms
        public var mMunjaState: Int
    }

    public struct PushAppImgDataReq: Codable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mImgId = "iId"
            case mMsgId = "mId"
        }

        public var mCid: Int = UmsProtocol.APP_IMG_DATA_CMD_ID
        public var mAccountId: String
        public var mImgId: String
        public var mMsgId: String
    }

    public struct PushAppImgDataRes: Decodable {
        enum CodingKeys: String, CodingKey {
            case mCid = "cId"
            case mAccountId = "aId"
            case mResultCode = "rc"
            case mComment = "cmt"
            case mImgData = "imgD"
        }

        public var mCid: Int = UmsProtocol.APP_IMG_DATA_CMD_ID
        public var mAccountId: String
        public var mResultCode: Int
        public var mComment: String?

        public var mImgData: String
    }
}
